model-optimizer/extensions/ops/dft.py

```diff
@@ -0,0 +1,133 @@
+# Copyright (C) 2018-2021 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+
+from mo.front.common.partial_infer.utils import int64_array
+from mo.graph.graph import Node, Graph
+from mo.ops.op import Op
+
+
+class FFTBase(Op):
+    enabled = False
+    op = None
+    version = 'opset7'
+
+    def __init__(self, graph: Graph, attrs: dict):
+        mandatory_props = {
+            'out_ports_count': 1,
+            'in_ports_count': 3,
+            'version': self.version,
+            'infer': self.infer
+        }
+        super().__init__(graph, mandatory_props, attrs)
+
+    def infer(self, node: Node):
+        node_name = node.soft_get(node.name, node.id)
+        assert len([p for p in node.in_ports().values() if not p.disconnected()]) in [2, 3], \
+            '(I)DFT node {} must have 2 or 3 inputs'.format(node_name)
+
+        src_shape = node.in_port(0).data.get_shape()
+        assert src_shape is not None, 'The input data shape of (I)DFT node {} must not be None'.format(node_name)
+        assert src_shape[-1] == 2, \
+            'The last dimension of input shape of (I)DFT node {} should be equal to 2'.format(node_name)
+
+        input_rank = len(src_shape)
+        assert input_rank >= 2, 'The input rank of (I)DFT node {} should be greater or equal to 2'.format(node_name)
+
+        axes = FFTBase.get_axes(node)
+        assert input_rank >= len(axes) + 1, \
+            'The input rank must be greater than number of (I)DFT node {} axes'.format(node_name)
+        axes = FFTBase.canonicalize_axes(axes, input_rank)
+        assert (input_rank - 1) not in axes, '(I)DFT node {} axes cannot contain the last axis'.format(node_name)
+        assert len(set(axes)) == len(axes), '(I)DFT node {} axes must be unique.'.format(node_name)
+
+        output_shape = int64_array(src_shape)
+        if node.is_in_port_connected(2):
+            signal_size = FFTBase.get_signal_size(node)
+            signal_size = FFTBase.canonicalize_signal_size(signal_size, axes, src_shape)
```

Comment:
The same comment as above for the signal size.

Reply:
We have no any extractors for PyTorch models, if these models were not converted in the ONNX format. Next, this canonicalization is a part of our DFT and IDFT specifications. Moreover, nGraph operations DFT and IDFT support -1s in `signal_size` input. Also see CPU implementation of DFT and IDFT in the PR #5383.